src/search.js

```diff
@@ -1,10 +1,15 @@
-import {ascend, assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy, sortWith, partition} from "ramda"
+import {ascend, assoc, concat, filter, isEmpty, map, pipe, prop, sortBy, sortWith, partition} from "ramda"
 import * as diacritics from 'diacritics'
 
+function preprocessParameters(parameters) {
+  return Object.keys(parameters).map(
+    paramId => Object.assign(parameters[paramId], {'name': paramId})
```

Comment:
Warning: this form of `Object.assign` mutates its first argument (`parameters[paramId]`).

Instead, use `Object.assign({}, parameters[paramId], {'name': paramId})`

Or use `assoc` from Ramda, like this: `assoc(parameters[paramId], {'name': paramId})` 😄